File: vnes-emulator/src/main/java/vnes/emulator/ChannelRegistry.java
package vnes.emulator;

import vnes.emulator.channels.IChannel;

import java.util.HashMap;
import java.util.Map;

public class ChannelRegistry {
    private final Map<Integer, IChannel> addressToChannelMap = new HashMap<>();

    public void registerChannel(int startAddr, int endAddr, IChannel channel) {
        for (int addr = startAddr; addr <= endAddr; addr++) {
            addressToChannelMap.put(addr, channel);
        }
    }

    public IChannel getChannel(int address) {
        return addressToChannelMap.get(address);
    }
}

File: knes-emulator/src/main/kotlin/knes/emulator/papu/ChannelRegistry.kt
package knes.emulator.papu

class ChannelRegistry {
    private val addressToChannelMap: MutableMap<Int?, PAPUChannel?> = HashMap<Int?, PAPUChannel?>()

    fun registerChannel(startAddr: Int, endAddr: Int, channel: PAPUChannel?) {
        for (addr in startAddr..endAddr) {
            addressToChannelMap.put(addr, channel)
        }
    }

    fun getChannel(address: Int): PAPUChannel? {
        return addressToChannelMap.get(address)
    }
}